config.py

```diff
@@ -384,6 +384,8 @@ def setup_extension_modules(self):
     if self.debug_mode:
       for k in self.compiler_flags:
         self.compiler_flags[k].append('-g')
+        ind = [i for i, item in enumerate(self.compiler_flags[k]) if item.startswith('-O')]
+        self.compiler_flags[k][ind[0]] = '-O0'
```

Comment:
-O0 is an option. We should consider -Og. It seems like a good compromise if what I read on the link below is correct. https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

Reply:
We are not the compiler experts. It is better to believe the official doc. However, if any problems in the future, just revert to -O0.